### PR TITLE
Dev spring boot order

### PR DESCRIPTION
**Pull request by OlegTuryk**

### CODE REVIEW

File: src/main/java/com/app/onlinebookstore/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+@Tag(name = "Order management",
+        description = "Endpoints for managing orders and orderItems")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (JeniaSan) [inline src/main/java/com/app/onlinebookstore/controller/OrderController.java:30]: check role "user" is redundant. Please, remove it everywhere

File: src/main/java/com/app/onlinebookstore/repository/OrderItemRepository.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    List<OrderItem> findOrderItemsByOrderIdAndUserId(Long orderId, Long userId, Pageable pageable);
+
+    @Query("""
+            FROM OrderItem oi
+            WHERE oi.order.id = :orderId
+            AND oi.order.user.id = :userId
+            AND oi.id = :orderItemId
+            """)
+    Optional<OrderItem> findOrderItemByOrderIdAndUserIdAndOrderItemId(
+            Long orderId, Long orderItemId, Long userId);
```
Comment (JeniaSan) [inline src/main/java/com/app/onlinebookstore/repository/OrderItemRepository.java:28]: Too many redundant params in this request. You can do this logic in service

**Review by JeniaSan — CHANGES_REQUESTED** [review]
Good job, but some minor mistakes

File: src/main/java/com/app/onlinebookstore/controller/OrderController.java (hunk 1 above)
Comment (OlegTuryk) [reply src/main/java/com/app/onlinebookstore/controller/OrderController.java:30]: about that someone talk to use it like in shoppingCart

File: src/main/java/com/app/onlinebookstore/repository/OrderItemRepository.java (hunk 2 above)
Comment (OlegTuryk) [reply src/main/java/com/app/onlinebookstore/repository/OrderItemRepository.java:28]: I think it not a problem

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**